Updating from 2023 json to SQL file throws error on deserializing April Tag pipeline

In #981, the enum `TargetModel.k6in_16h5` was renamed to `kAprilTag6in_16h5`. When trying to update the settings stored in the JSON files, this causes the deserialization of the pipeline to fail and the pipeline reverts to reflective. 

** Exception **
```
[2023-12-17 11:17:33] [General - LegacyConfigProvider] [ERROR] Exception while deserializing cameras/Arducam_OV9281_USB_Camera/pipelines/New Pipeline.json: Cannot deserialize value of type `org.photonvision.vision.target.TargetModel` from String "k6in_16h5": not one of the values accepted for Enum class: [k2016HighGoal, kAprilTag6in_16h5, kAprilTag6p5in_36h11, k2019DualTarget, k2020HighGoalOuter, k2022CircularCargoBall, kCircularPowerCell7in]
 at [Source: (File); line: 66, column: 19] (through reference chain: org.photonvision.vision.pipeline.AprilTagPipelineSettings["targetModel"])
[2023-12-17 11:17:33] [General - LegacyConfigProvider] [ERROR] com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `org.photonvision.vision.target.TargetModel` from String "k6in_16h5": not one of the values accepted for Enum class: [k2016HighGoal, kAprilTag6in_16h5, kAprilTag6p5in_36h11, k2019DualTarget, k2020HighGoalOuter, k2022CircularCargoBall, kCircularPowerCell7in]
 at [Source: (File); line: 66, column: 19] (through reference chain: org.photonvision.vision.pipeline.AprilTagPipelineSettings["targetModel"])
        at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:67)
        at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:2002)
        at com.fasterxml.jackson.databind.DeserializationContext.handleWeirdStringValue(DeserializationContext.java:1230)
        at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._deserializeAltString(EnumDeserializer.java:415)
        at com.fasterxml.jackson.databind.deser.std.EnumDeserializer._fromString(EnumDeserializer.java:279)
        at com.fasterxml.jackson.databind.deser.std.EnumDeserializer.deserialize(EnumDeserializer.java:248)
        at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:314)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.jsontype.impl.AsArrayTypeDeserializer._deserialize(AsArrayTypeDeserializer.java:120)
        at com.fasterxml.jackson.databind.jsontype.impl.AsArrayTypeDeserializer.deserializeTypedFromObject(AsArrayTypeDeserializer.java:61)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeWithType(BeanDeserializerBase.java:1296)
        at com.fasterxml.jackson.databind.deser.impl.TypeWrappedDeserializer.deserialize(TypeWrappedDeserializer.java:74)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4825)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3637)
        at org.photonvision.common.util.file.JacksonUtils.deserialize(JacksonUtils.java:88)
        at org.photonvision.common.configuration.LegacyConfigProvider.lambda$loadCameraConfigs$2(LegacyConfigProvider.java:339)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at org.photonvision.common.configuration.LegacyConfigProvider.loadCameraConfigs(LegacyConfigProvider.java:351)
        at org.photonvision.common.configuration.LegacyConfigProvider.load(LegacyConfigProvider.java:213)
        at org.photonvision.common.configuration.ConfigManager.translateLegacyIfPresent(ConfigManager.java:96)
        at org.photonvision.common.configuration.ConfigManager.load(ConfigManager.java:183)
        at org.photonvision.Main.main(Main.java:348)
```

--- a/photon-core/src/main/java/org/photonvision/vision/target/TargetModel.java
+++ b/photon-core/src/main/java/org/photonvision/vision/target/TargetModel.java
@@ -16,6 +16,7 @@
  */
 package org.photonvision.vision.target;
 
+import com.fasterxml.jackson.annotation.JsonAlias;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
@@ -108,6 +109,7 @@ public enum TargetModel implements Releasable {
                             -Units.inchesToMeters(9.5) / 2)),
             0),
     // 2023 AprilTag, with 6 inch marker width (inner black square).
+    @JsonAlias({"k6in_16h5"})
     kAprilTag6in_16h5(
             // Corners of the tag's inner black square (excluding white border)
             List.of(
@@ -117,6 +119,7 @@ public enum TargetModel implements Releasable {
                     new Point3(Units.inchesToMeters(3), -Units.inchesToMeters(3), 0)),
             Units.inchesToMeters(3 * 2)),
     // 2024 AprilTag, with 6.5 inch marker width (inner black square).
+    @JsonAlias({"k6p5in_36h11"})
     kAprilTag6p5in_36h11(
             // Corners of the tag's inner black square (excluding white border)
             List.of(

--- a/photon-core/src/main/java/org/photonvision/vision/target/TargetModel.java
+++ b/photon-core/src/main/java/org/photonvision/vision/target/TargetModel.java
@@ -119,7 +119,7 @@ public enum TargetModel implements Releasable {
                     new Point3(Units.inchesToMeters(3), -Units.inchesToMeters(3), 0)),
             Units.inchesToMeters(3 * 2)),
     // 2024 AprilTag, with 6.5 inch marker width (inner black square).
-    @JsonAlias({"k6p5in_36h11"})
+    @JsonAlias({"k6p5in_36h11", "k200mmAprilTag", "kAruco6p5in_36h11"})
     kAprilTag6p5in_36h11(
             // Corners of the tag's inner black square (excluding white border)
             List.of(

--- a/photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java
+++ b/photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java
@@ -31,6 +31,7 @@ import org.photonvision.common.logging.Logger;
 import org.photonvision.common.util.TestUtils;
 import org.photonvision.common.util.file.JacksonUtils;
 import org.photonvision.vision.pipeline.AprilTagPipelineSettings;
+import org.photonvision.vision.pipeline.CVPipelineSettings;
 import org.photonvision.vision.pipeline.ColoredShapePipelineSettings;
 import org.photonvision.vision.pipeline.ReflectivePipelineSettings;
 import org.photonvision.vision.target.TargetModel;
@@ -138,6 +139,20 @@ public class ConfigTest {
         writer.close();
         Assertions.assertDoesNotThrow(
                 () -> JacksonUtils.deserialize(Path.of("test.json"), CameraConfiguration.class));
+
+        new File("test.json").delete();
+    }
+
+    @Test
+    public void testJacksonHandlesOldTargetEnum() throws IOException {
+        var str =
+                "[ \"AprilTagPipelineSettings\", {\n" + "  \"targetModel\" : \"k6in_16h5\"\n" + "} ]\n";
+        var writer = new FileWriter("test.json");
+        writer.write(str);
+        writer.flush();
+        writer.close();
+        Assertions.assertDoesNotThrow(
+                () -> JacksonUtils.deserialize(Path.of("test.json"), CVPipelineSettings.class));
 
         new File("test.json").delete();
     }

--- a/photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java
+++ b/photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java
@@ -133,27 +133,31 @@ public class ConfigTest {
     public void testJacksonHandlesOldVersions() throws IOException {
         var str =
                 "{\"baseName\":\"aaaaaa\",\"uniqueName\":\"aaaaaa\",\"nickname\":\"aaaaaa\",\"FOV\":70.0,\"path\":\"dev/vid\",\"cameraType\":\"UsbCamera\",\"currentPipelineIndex\":0,\"camPitch\":{\"radians\":0.0},\"calibrations\":[], \"cameraLEDs\":[]}";
-        var writer = new FileWriter("test.json");
+        File tempFile = File.createTempFile("test", ".json");
+        tempFile.deleteOnExit();
+        var writer = new FileWriter(tempFile);
         writer.write(str);
         writer.flush();
         writer.close();
         Assertions.assertDoesNotThrow(
-                () -> JacksonUtils.deserialize(Path.of("test.json"), CameraConfiguration.class));
+                () -> JacksonUtils.deserialize(tempFile.toPath(), CameraConfiguration.class));
 
-        new File("test.json").delete();
+        tempFile.delete();
     }
 
     @Test
     public void testJacksonHandlesOldTargetEnum() throws IOException {
-        var str =
-                "[ \"AprilTagPipelineSettings\", {\n" + "  \"targetModel\" : \"k6in_16h5\"\n" + "} ]\n";
-        var writer = new FileWriter("test.json");
+        var str = "[ \"AprilTagPipelineSettings\", {\n  \"targetModel\" : \"k6in_16h5\"\n} ]\n";
+
+        File tempFile = File.createTempFile("test", ".json");
+        tempFile.deleteOnExit();
+        var writer = new FileWriter(tempFile);
         writer.write(str);
         writer.flush();
         writer.close();
         Assertions.assertDoesNotThrow(
-                () -> JacksonUtils.deserialize(Path.of("test.json"), CVPipelineSettings.class));
+                () -> JacksonUtils.deserialize(tempFile.toPath(), CVPipelineSettings.class));
 
-        new File("test.json").delete();
+        tempFile.delete();
     }
 }

--- a/photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java
+++ b/photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java
@@ -155,8 +155,11 @@ public class ConfigTest {
         writer.write(str);
         writer.flush();
         writer.close();
-        Assertions.assertDoesNotThrow(
-                () -> JacksonUtils.deserialize(tempFile.toPath(), CVPipelineSettings.class));
+
+        AprilTagPipelineSettings settings =
+                (AprilTagPipelineSettings)
+                        JacksonUtils.deserialize(tempFile.toPath(), CVPipelineSettings.class);
+        Assertions.assertEquals(TargetModel.kAprilTag6in_16h5, settings.targetModel);
 
         tempFile.delete();
     }

